cli/popper/commands/cmd_run.py

```diff
@@ -120,124 +112,148 @@
     default=popper.scm.get_git_root_folder()
 )
 @pass_context
-def cli(ctx, action, wfile, skip_clone, skip_pull, skip, reuse,
-        recursive, quiet, debug, dry_run, parallel, log_file,
-        with_dependencies, on_failure, runtime, workspace):
-    """Execute github actions workflows.
+def cli(ctx, **kwargs):
+    """Executes one or more pipelines and reports on their status.
     """
-    popper.scm.get_git_root_folder()
+    if os.environ.get('CI') == 'true':
+        # When CI is set,
+        log.info('Running in CI environment...')
+        occurences = parse_commit_message()
+        if occurences:
+            for params in get_params(occurences):
+                kwargs.update(params)
+                if kwargs['recursive']:
+                    log.warn('When CI is set, --recursive is ignored.')
+                    kwargs['recursive'] = False
+                prepare_pipeline(**kwargs)
+        else:
+            # If no special keyword is found, we run all the workflows,
+            # recursively.
+            kwargs['recursive'] = True
+            prepare_pipeline(**kwargs)
+    else:
+        # When CI is not set,
+        prepare_pipeline(**kwargs)
+
+
+def prepare_pipeline(**kwargs):
```

Comment:
we might want to rename this to `prepare_workflow_execution` to be consistent with terminology